codersdk/workspaceagents.go

```diff
@@ -35,15 +35,15 @@ const (
 )
 
 type WorkspaceAgent struct {
-	ID                   uuid.UUID            `json:"id"`
-	CreatedAt            time.Time            `json:"created_at"`
-	UpdatedAt            time.Time            `json:"updated_at"`
-	FirstConnectedAt     *time.Time           `json:"first_connected_at,omitempty"`
-	LastConnectedAt      *time.Time           `json:"last_connected_at,omitempty"`
-	DisconnectedAt       *time.Time           `json:"disconnected_at,omitempty"`
-	Status               WorkspaceAgentStatus `json:"status"`
+	ID                   uuid.UUID            `json:"id" format:"uuid"`
+	CreatedAt            time.Time            `json:"created_at" format:"date-time"`
```

Comment:
Perhaps I've asked before, but is there a way to automatically transform these based on the Go type? It seems unfortunate we'd have to tag every new struct, and we will almost certainly miss to add the tag to fields in the future.

It'd be nice if `format` was reserved as an override since `uuid` and `time.Time` are essentially never a different format (would take a custom json Marshaller at which point using `format` would be appropriate).

Reply:
I'm afraid that we would need a fork to apply this feature, let's say, `auto-format`. It doesn't look like something we can truly solve just with validation.

 I'm not sure why it hasn't been implemented this way. Even official [examples](https://github.com/swaggo/swag/blob/f6170518d62292b434f80cb109f585a972b6cb42/example/celler/model/account.go#L14) suggest using `format` tags.

Reply:
I can certainly understand why it's not the default (due to middlewares and handlers that can do whatever). But it'd be nice to enable such a feature.

Do you think they'd be open to it if we made a PR? Seems like many people could benefit.

Reply:
At the moment, we're using version `swaggo v1.8.6`, but I can try to bump it to `v1.8.9` so that we will also benefit from future PRs. AFAIR v1.8.9 added a lot of noise around all possible values of `time.Duration` or strange value examples:

```diff
-    "value": 0
+    "value": -9223372036854776000
```

Reply:
Ah, if `v1.8.9` is problematic we can punt it for now unless you think there's a benefit to upgrading?

Reply:
Let me prepare a draft PR and we can discuss the upgrade there.

Reply:
https://github.com/coder/coder/pull/5590